lib/iolog/iolog_fileio.c

```diff
@@ -1061,7 +1061,6 @@ iolog_write_info_file_json(int dfd, const char *parent, struct iolog_info *info)
             "unable to open %s/log.json", parent);
         goto done;
     }
-    fd = -1;
```

Comment:
We still need to set fd to -1 to avoid closing the fd twice but this should either happen after the fchown() or fchown() should use fileno(fp) instead.

Reply:
I have moved it down a bit.